resources/js/components/ui/Table/TQueryTable.tsx

```diff
@@ -61,51 +43,77 @@ export interface TQueryTableProps {
    * - embedded - the table is displayed along with other elements in a page, typically with not
    * many rows, without sticky elements.
    */
-  mode: DisplayMode;
+  readonly mode: DisplayMode;
   /** The prefix used for the data query (this allows invalidating the tquery data). */
-  staticPrefixQueryKey: readonly unknown[];
+  readonly staticPrefixQueryKey: readonly unknown[];
   /** The entity URL, must not change. */
-  staticEntityURL: string;
-  staticTranslations?: TableTranslations;
+  readonly staticEntityURL: string;
+  readonly staticTranslations?: TableTranslations;
   /**
    * The filter that is always applied to the data, regardless of other filtering.
    * This is used to create e.g. a table of entities A on the details page of a particular
    * entity B, so only entities A related direclty to that particular entity B should be shown.
    */
-  intrinsicFilter?: FilterH;
+  readonly intrinsicFilter?: FilterH;
+  /** The definition of the columns in the table, in their correct order. */
+  readonly columns: readonly PartialColumnConfig[];
+  readonly initialSort?: SortingState;
+  readonly initialPageSize?: number;
+  /** Element to put below table, after the summary. */
+  readonly customSectionBelowTable?: JSX.Element;
+}
+
+export interface PartialColumnConfig {
+  /** The name (id) of the column. */
+  readonly name: string;
   /**
-   * A list of columns that are always included in the request (and available on the row objects),
-   * even if they are not visible.
+   * Whether this column has a corresponding tquery column (with the same name) that it shows.
+   * Default: true.
    */
-  intrinsicColumns?: readonly string[];
-  /** Additional column names. Their options are taken from `columnOptions`. */
-  additionalColumns?: readonly string[];
-  /** Overrides for the definition of specific columns. */
-  columnOptions?: Readonly<Partial<Record<string, ColumnOptions>>>;
+  readonly isDataColumn?: boolean;
+  /** Additional columns from the tquery row that are used to construct the displayed value. */
+  readonly extraDataColumns?: readonly ColumnName[];
   /**
-   * The ordering of the columns. All the columns present on the backend and not present
-   * in this list are placed at the end.
-   *
-   * This list can be used to sort both data-based columns and additional columns
-   * (declared in additionalColumns).
-   *
-   * In the current implementation the order of columns cannot be changed.
+   * The TanStack column definition. If dataColumn is true, the tquery column is displayed by
+   * default. Otherwise, columnDef needs to be specified to display anything.
+   * Only dataColumn and extraDataColumns can be used in the columnDef.
```

Comment:
dataColumn -> name ?

"can be used in the columnDef" -> chodzi o cell function czy o coś jeszcze?

to drugie fajnie jakby accessorFn działał tak, że co masz w dataColumns dostajesz w c.getValue() i nie używasz c.row.getValue()

Reply:
Chyba tylko cell, ale w sumie kto tam wie co jest w tym columnDef...

Jest jeszcze c.row.original na przykład, i do tego część tego to są story, no więc byłoby fajnie ale na razie to zostawiam bo nie mam pomysłu jak to elegancko ogarnąć. Chociaż oczywiście jest error-prone.